stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/StringExtensions.kt

```diff
@@ -57,3 +65,150 @@ public fun String.cidToTypeAndId(): Pair<String, String> {
+ * @param resizeMode Sets the image resizing mode. The default mode is [StreamCdnResizeImageMode.CLIP].
+ * @param cropMode Sets the image crop mode. The default mode is [StreamCdnCropImageMode.CENTER].
+ */
+public fun String.createResizedStreamCdnImageUrl(
+    @FloatRange(from = 0.0, to = 1.0, fromInclusive = false) resizedWidthPercentage: Float,
+    @FloatRange(from = 0.0, to = 1.0, fromInclusive = false) resizedHeightPercentage: Float,
+    resizeMode: StreamCdnResizeImageMode? = null,
+    cropMode: StreamCdnCropImageMode? = null,
+): String {
+    val logger = StreamLog.getLogger("Chat:resizedStreamCdnImageUrl")
+
+    val streamCdnImageDimensions = this.getStreamCdnHostedImageDimensions()
+
+    return if (streamCdnImageDimensions != null) {
+
+        val imageLinkUri = this.toUri()
+
+        if (imageLinkUri.wasImagePreviouslyResized()) {
+            logger.w {
+                "Image URL already contains resizing parameters. Please apply resizing parameters only to " +
+                    "original image URLs."
+            }
+
+            return this@createResizedStreamCdnImageUrl
+        }
+
+        val resizedWidth: Int = (streamCdnImageDimensions.originalWidth * resizedWidthPercentage).toInt()
+        val resizedHeight: Int = (streamCdnImageDimensions.originalHeight * resizedHeightPercentage).toInt()
+
+        val resizedImageUrl = imageLinkUri
+            .buildUpon()
+            .appendValueAsQueryParameterIfNotNull(
+                key = StreamCdnResizeImageQueryParameterKeys.QUERY_PARAMETER_KEY_RESIZED_WIDTH,
+                value = resizedWidth
+            )
+            .appendValueAsQueryParameterIfNotNull(
+                key = StreamCdnResizeImageQueryParameterKeys.QUERY_PARAMETER_KEY_RESIZED_HEIGHT,
+                value = resizedHeight
+            )
+            .appendValueAsQueryParameterIfNotNull(
+                key = StreamCdnResizeImageQueryParameterKeys.QUERY_PARAMETER_KEY_RESIZE_MODE,
+                value = resizeMode?.queryParameterName
+            )
+            .appendValueAsQueryParameterIfNotNull(
+                key = StreamCdnResizeImageQueryParameterKeys.QUERY_PARAMETER_KEY_CROP_MODE,
+                value = cropMode?.queryParameterName
+            )
+            .build()
+            .toString()
+
+        logger.i {
+            "Resized Stream CDN hosted image URL: $resizedImageUrl"
+        }
+
+        resizedImageUrl
+    } else {
+        logger.i {
+            "Image not hosted by Stream's CDN or not containing original width and height query parameters " +
+                "was not resized"
+        }
+        this
+    }
+}
+
+/**
+ * Checks if a string contains resizing related query parameters.
+ *
+ * @return true if the URL contains resizing parameters, false otherwise.
+ */
+private fun Uri.wasImagePreviouslyResized(): Boolean =
+    queryParameterNames.any { queryParam ->
```

Comment:
Maybe we can do something like:

`queryParameterNames.intersect(listOf(ourParameters)).isNotEmpty()`

Reply:
It scales better if we add more params as we won't have to check every single one

Reply:
Implemented the suggestion in the following commit: https://github.com/GetStream/stream-chat-android/pull/4600/commits/5c833a3550d73556dfb47913afe7a479230b4859